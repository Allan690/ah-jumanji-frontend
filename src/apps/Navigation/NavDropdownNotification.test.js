import React from 'react';

import Enzyme, { shallow, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

import { MemoryRouter } from 'react-router-dom';
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import DropDownNotification from './NavDropDownNotification';

Enzyme.configure({ adapter: new Adapter() });
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('<DropDownNotification />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<DropDownNotification debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('DropDownNotification', () => {
  it('should render self and subcomponents', () => {
    const store = mockStore({});
    const enzymeWrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <DropDownNotification />
        </MemoryRouter>
      </Provider>,);
    expect(enzymeWrapper.find('DropDownNotification')).toBeDefined();
    expect(enzymeWrapper.find('img')).toHaveLength(1);
  });
});
