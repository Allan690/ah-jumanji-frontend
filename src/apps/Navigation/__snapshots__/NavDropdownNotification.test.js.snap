// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDownNotification /> should match the snapshot 1`] = `
<li
  className="dropdown-item"
>
  <img
    alt="follower avatar"
    className="followers-avatar rounded-circle"
  />
  <button
    className="btn-link"
    type="button"
  >
     
  </button>
   
  <span
    className="float-right"
  >
    <button
      className="btn-sm btn-link"
      type="button"
    >
      <i
        className="fas fa-check-circle"
      />
      Mark As Read
    </button>
  </span>
  <br />
  <i
    className="text-muted"
  >
    <span
      className="fas fa-clock "
    />
     
     ago
  </i>
</li>
`;
